pinot-minion/src/main/java/org/apache/pinot/minion/MinionStarter.java

```diff
@@ -284,7 +284,8 @@ private void addInstanceTagIfNeeded() {
     HelixAdmin helixAdmin = _helixManager.getClusterManagmentTool();
     String clusterName = _helixManager.getClusterName();
     InstanceConfig instanceConfig = helixAdmin.getInstanceConfig(clusterName, _instanceId);
-    if (instanceConfig.getTags().isEmpty()) {
+    List<String> instanceTags = instanceConfig.getTags();
```

Comment:
The returned instanceConfig could be null if record is null. @amrishlal Please add the check `instanceConfig == null ` as well.

```
  public static <T extends HelixProperty> T convertToTypedInstance(Class<T> clazz, ZNRecord record) {
    if (record == null) {
      return null;
    }

    try {
      Constructor<T> constructor = clazz.getConstructor(ZNRecord.class);
      return constructor.newInstance(record);
    } catch (Exception e) {
      LOG.error("Exception convert znrecord: " + record + " to class: " + clazz, e);
    }

    return null;
  }
```